server/src/Metrics.hs

```diff
@@ -1,31 +1,27 @@
 module Metrics where
 
 import           Prometheus
+import qualified Network.HTTP.Types as Http
+import qualified Data.ByteString.Char8 as BS8
+
+type RequestCounter = Vector (String, String) Counter
 
 data IcepeakMetrics = IcepeakMetrics
-  { icepeakMetricsPutCounter      :: Metric Counter
-  , icepeakMetricsGetCounter      :: Metric Counter
-  , icepeakMetricsDeleteCounter   :: Metric Counter
+  { icepeakMetricsRequestCounter  :: Metric RequestCounter
   , icepeakMetricsDataSize        :: Metric Gauge
   , icepeakMetricsSubscriberCount :: Metric Gauge
   }
 
 createAndRegisterIcepeakMetrics :: IO IcepeakMetrics
 createAndRegisterIcepeakMetrics = IcepeakMetrics
-  <$> registerIO (counter (Info "put_requests" "Total number of PUT requests since starting Icepeak."))
-  <*> registerIO (counter (Info "get_requests" "Total number of GET requests since starting Icepeak."))
-  <*> registerIO (counter (Info "delete_requests" "Total number of DELETE requests since starting Icepeak."))
-  <*> registerIO (gauge (Info "data_size" "Size of data file in bytes."))
-  <*> registerIO (gauge (Info "subscriber_count" "Number of websocket subscriber connections."))
-
-notifyGetRequest :: IcepeakMetrics -> IO ()
-notifyGetRequest = incCounter . icepeakMetricsGetCounter
-
-notifyPutRequest :: IcepeakMetrics -> IO ()
-notifyPutRequest = incCounter . icepeakMetricsPutCounter
-
-notifyDeleteRequest :: IcepeakMetrics -> IO ()
-notifyDeleteRequest = incCounter . icepeakMetricsDeleteCounter
+  <$> registerIO (vector ("method", "status") $ counter (Info "icepeak_http_requests" "Total number of HTTP requests since starting Icepeak."))
+  <*> registerIO (gauge (Info "icepeak_data_size" "Size of data file in bytes."))
```

Comment:
Can you make it shorter?

``` haskell
createAndRegisterIcepeakMetrics = IcepeakMetrics
  <$> registerIO (vector ("method", "status") 
        $ counter (Info "icepeak_http_requests" "Total number of HTTP requests since starting Icepeak."))
  <*> registerIO (gauge (Info "icepeak_data_size" "Size of data file in bytes."))
  <*> registerIO (gauge (Info "icepeak_subscriber_count" "Number of websocket subscriber connections."))
```

or 

``` haskell
createAndRegisterIcepeakMetrics = IcepeakMetrics
  <$> registerIO (vector ("method", "status")  requestCounter
  <*> registerIO (gauge (Info "icepeak_data_size" "Size of data file in bytes."))
  <*> registerIO (gauge (Info "icepeak_subscriber_count" "Number of websocket subscriber connections."))
    where requestCounter = counter (Info "icepeak_http_requests" "Total number of HTTP requests since starting Icepeak."))
```